### PR TITLE
Warning status

### PR DESCRIPTION
**Pull request by werdes72**
**Description**

Changes proposed in this pull request:
- Add warning status to Istio module

**Related issue(s)**
https://github.com/kyma-project/lifecycle-manager/issues/525

### CODE REVIEW

File: internal/status/status.go [hunk 1]
```diff
@@ -60,3 +61,9 @@ func (DefaultStatusHandler) SetDeleting(ctx context.Context, client client.Clien
 	istioCR.Status.State = operatorv1alpha1.Deleting
 	return update(ctx, client, istioCR, condition, retryTime...)
 }
+
+func (DefaultStatusHandler) SetWarning(ctx context.Context, err described_errors.DescribedError, client client.Client, istioCR *operatorv1alpha1.Istio, condition metav1.Condition, retryTime ...time.Duration) (ctrl.Result, error) {
```
Comment (barchw) [inline internal/status/status.go:65]: As warning is also a `described_error` I think that moving the logic under `SetError` func would make more sense (so we should set `istioCR.Status.State` based on err.level()). Then the new logic in https://github.com/kyma-project/istio/blob/fa38c94d0718c4153f1d2adf55452206d3b0edba/controllers/istio_controller.go#L84 could be removed, only leaving SetError at the end.

File: internal/reconciliations/proxy/sidecars_test.go [hunk 2]
```diff
@@ -141,3 +141,7 @@ func (s StatusMock) SetError(_ context.Context, _ described_errors.DescribedErro
 func (s StatusMock) SetDeleting(_ context.Context, _ client.Client, _ *operatorv1alpha1.Istio, _ metav1.Condition, _ ...time.Duration) (ctrl.Result, error) {
 	return s.result, s.err
 }
+
```
Comment (barchw) [inline internal/reconciliations/proxy/sidecars_test.go:144]: Maybe the name for `described_errors.DescribedError` should be adjusted (e.g. `described_errors.DescribedState`)?

**Review by barchw — CHANGES_REQUESTED** [review]
Unit test coverage is missing

**Review by barchw — DISMISSED** [review]

File: docs/user/01-20-istio-custom-resource.md [hunk 3]
```diff
@@ -37,6 +37,6 @@ This table lists all the possible parameters of the given resource together with
 
 **Status:**
 
-| Parameter | Type | Description |
-| ---- | ----------- | ---- |
-| **state** (required) | string | Signifies the current state of **CustomObject**. Its value can be either `Ready`, `Processing`, `Error`, or `Deleting`. |
\ No newline at end of file
+| Parameter | Type | Description                                                                                                                       |
+| ---- | ----------- |-----------------------------------------------------------------------------------------------------------------------------------|
+| **state** (required) | string | Signifies the current state of **CustomObject**. Its value can be either `Ready`, `Processing`, `Error`, `Warning` or `Deleting`. |
```
Comment (nataliasitko) [inline docs/user/01-20-istio-custom-resource.md:42]:
```suggestion
| **state** (required) | string | Signifies the current state of **CustomObject**. Its value can be either `Ready`, `Processing`, `Error`, `Warning`, or `Deleting`. |
```

**Review by nataliasitko — APPROVED** [review]

**Review by barchw — APPROVED** [review]

**Pull request merged**